posts/2019-12-19-Rust-1.40.0.md

```diff
@@ -0,0 +1,246 @@
+[`Option::flatten`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.flatten
+[`Option::as_deref`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref
+[`Option::as_deref_mut`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref_mut
+[`UdpSocket::peer_addr`]: https://doc.rust-lang.org/std/net/struct.UdpSocket.html#method.peer_addr
+[`{f32,f64}::to_be_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.to_be_bytes
+[`{f32,f64}::to_le_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.to_le_bytes
+[`{f32,f64}::to_ne_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.to_ne_bytes
+[`{f32,f64}::from_be_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.from_be_bytes
+[`{f32,f64}::from_le_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.from_le_bytes
+[`{f32,f64}::from_ne_bytes`]: https://doc.rust-lang.org/std/primitive.f32.html#method.from_ne_bytes
+
+[`Option::take`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.take
+[`Cell::take`]: https://doc.rust-lang.org/std/cell/struct.Cell.html#method.take
+[`mem::replace`]: https://doc.rust-lang.org/std/mem/fn.replace.html
+[`unimplemented!()`]: https://doc.rust-lang.org/std/macro.unimplemented.html
+[`Option::flatten`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.flatten
+[`Option::as_ref`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref
+[`Option::as_mut`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.as_mut
+[`Deref`]: https://doc.rust-lang.org/std/ops/trait.Deref.html
+[`DerefMut`]: https://doc.rust-lang.org/std/ops/trait.DerefMut.html
+[`Iterator::flatten`]: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flatten 
+
+In Rust 1.40.0 the following functions and macros were stabilized:
+
+- [`todo!`]
+
+  A macro, which is a shorter, more memorable, and convenient version of [`unimplemented!()`].
+
+- [`slice::repeat`]
+
+  Creates a `Vec<T>` by repeating a slice `n` times.
+
+- [`mem::take`]
+
+  This function `take`s the value out of a mutable reference and replaces it with the type's default. This is similar to [`Option::take`] and [`Cell::take` ] and provides a convenient short-hand for [`mem::replace(&mut dst, Default::default())`][`mem::replace`].
+
+- [`BTreeMap::get_key_value`] and [`HashMap::get_key_value`]
+
+  Returns the key-value pair corresponding to the supplied key.
+
+- [`Option::as_deref`], [`Option::as_deref_mut`]
+
+  These work similarly to [`Option::as_ref`] and [`Option::as_mut`] but also use [`Deref`] and [`DerefMut`] respectively, so that `opt_box.as_deref()` and `opt_box.as_deref_mut()`, where `opt_box: Option<Box<T>>`, produce an `Option<&T>` and `Option<&mut T>` respectively.
+
+- [`Option::flatten`]
+
+  This function flatten an `Option<Option<T>>` to `Option<T>` producing `Some(x)` for `Some(Some(x))` and `None` otherwise. The function is similar to [`Iterator::flatten`].
```

Comment:
```suggestion
  This function flattens an `Option<Option<T>>` to `Option<T>` producing `Some(x)` for `Some(Some(x))` and `None` otherwise. The function is similar to [`Iterator::flatten`].
```